lang_python/parsing/ast_python.ml

```diff
@@ -253,7 +255,7 @@ type stmt =
   | While of expr (* test *) * stmt list (* body *) * stmt list (* orelse *)
   | If of expr (* test *) * stmt list (* body *) * stmt list (* orelse *)
   (* https://docs.python.org/2.5/whatsnew/pep-343.html *)
-  | With of expr (* context_expr *) * expr option (* optional_vars *) * stmt list (* body *)
+  | With of with_item list (* context items *) * stmt list (* body *)
```

Comment:
no need to change With. You can just desugar With X as A, Y as B into 2 With, With X as A (With Y as B (...))